collect_app/src/main/java/org/odk/collect/android/preferences/screens/AdminPreferencesFragment.java

```diff
@@ -127,79 +170,15 @@ private void displayPreferences(Fragment fragment) {
-            addPreferencesFromResource(R.xml.form_entry_access_preferences);
-
-            findPreference(KEY_MOVING_BACKWARDS).setOnPreferenceChangeListener((preference, newValue) -> {
-                if (((CheckBoxPreference) preference).isChecked()) {
-                    new MovingBackwardsDialog().show(getActivity().getSupportFragmentManager(), MOVING_BACKWARDS_DIALOG_TAG);
-                } else {
-                    SimpleDialog.newInstance(getActivity().getString(R.string.moving_backwards_enabled_title), 0, getActivity().getString(R.string.moving_backwards_enabled_message), getActivity().getString(R.string.ok), false).show(((AdminPreferencesActivity) getActivity()).getSupportFragmentManager(), SimpleDialog.COLLECT_DIALOG_TAG);
-                    onMovingBackwardsEnabled();
-                }
-                return true;
-            });
-            findPreference(KEY_JUMP_TO).setEnabled(settingsProvider.getAdminSettings().getBoolean(ALLOW_OTHER_WAYS_OF_EDITING_FORM));
-            findPreference(KEY_SAVE_MID).setEnabled(settingsProvider.getAdminSettings().getBoolean(ALLOW_OTHER_WAYS_OF_EDITING_FORM));
-        }
-
-        private void preventOtherWaysOfEditingForm() {
-            settingsProvider.getAdminSettings().save(ALLOW_OTHER_WAYS_OF_EDITING_FORM, false);
-            settingsProvider.getAdminSettings().save(KEY_EDIT_SAVED, false);
-            settingsProvider.getAdminSettings().save(KEY_SAVE_MID, false);
-            settingsProvider.getAdminSettings().save(KEY_JUMP_TO, false);
-            settingsProvider.getGeneralSettings().save(GeneralKeys.KEY_CONSTRAINT_BEHAVIOR, CONSTRAINT_BEHAVIOR_ON_SWIPE);
-
-            findPreference(KEY_JUMP_TO).setEnabled(false);
-            findPreference(KEY_SAVE_MID).setEnabled(false);
-
-            ((CheckBoxPreference) findPreference(KEY_JUMP_TO)).setChecked(false);
-            ((CheckBoxPreference) findPreference(KEY_SAVE_MID)).setChecked(false);
-        }
-
-        private void onMovingBackwardsEnabled() {
-            settingsProvider.getAdminSettings().save(ALLOW_OTHER_WAYS_OF_EDITING_FORM, true);
-            findPreference(KEY_JUMP_TO).setEnabled(true);
-            findPreference(KEY_SAVE_MID).setEnabled(true);
-        }
-    }
-
     public void preventOtherWaysOfEditingForm() {
-        FormEntryAccessPreferences fragment = (FormEntryAccessPreferences) getFragmentManager().findFragmentById(R.id.preferences_fragment_container);
+        FormEntryAccessPreferencesFragment fragment = (FormEntryAccessPreferencesFragment) getFragmentManager().findFragmentById(R.id.preferences_fragment_container);
         fragment.preventOtherWaysOfEditingForm();
     }
+
+    private void setProjectColorSummary() {
+        Spannable summary = new SpannableString("■");
+        summary.setSpan(new ForegroundColorSpan(Color.parseColor(currentProjectProvider.getCurrentProject().getColor())), 0, summary.length(), 0);
+        findPreference(PROJECT_COLOR_KEY).setSummary("");
```

Comment:
It might seem weird that I clear the summary first but it's a must. I'm not sure but I think that android uses some kind of caching and since the summary is always the same square just with different colors we need to clear it first.

Reply:
Now as I implemented SummaryProvider this is no longer the case.